krm-functions/configinject-fn/fn/function.go

```diff
@@ -139,84 +140,118 @@ func (f *FnR) desiredOwnedResourceList(forObj *fn.KubeObject) (fn.KubeObjects, e
+	// have the packagename and are in a repo that has deployment true
+	// The map will contain the latest published revision of the pr, if no pr
+	// is published it will have a reference to this pr
 	// we assume there needs to be 1 dependency that resolves
-	found := false
+	prmap := map[string]porchv1alpha1.PackageRevision{}
 	for _, pr := range prl.Items {
 		repo, ok := repomap[pr.Spec.RepositoryName]
 		if !ok {
 			return nil, fmt.Errorf("configinject repo name not found: %s", pr.Spec.RepositoryName)
 		}
-		// only analyse the packages with the packageName contained in the dependency requirement resource
-		// and only look at repo(s) where the deployment is true
-		// TBD: do we need to check the latest revision ?
+		// only analyse the packages with
+		// - the packageName contained in the dependency requirement resource
+		// - repo has deployment true
+		// - package is published
 		if pr.Spec.PackageName == depPackageName && repo.Spec.Deployment {
 			fn.Logf("configinject repo %s\n", pr.Spec.RepositoryName)
-			// get the package resources of the revision
-			prr := &porchv1alpha1.PackageRevisionResources{}
-			if err := f.Get(ctx, types.NamespacedName{Namespace: pr.Namespace, Name: pr.Name}, prr); err != nil {
-				return nil, err
-			}
-			// get the resource list from the package
-			rl, err := kptrl.GetResourceList(prr.Spec.Resources)
-			if err != nil {
-				return nil, err
-			}
-			// get the kptfile from the resourcelist to check condition status
-			kfko := rl.Items.GetRootKptfile()
-			if kfko == nil {
-				return nil, fmt.Errorf("mandatory Kptfile is missing from the package %s, repo %s", pr.Spec.PackageName, pr.Spec.RepositoryName)
-			}
-			kf := kptfilelibv1.KptFile{Kptfile: kfko}
 
-			// get the dependency objects in the package and check its status
-			gvk := schema.GroupVersionKind{
-				Group:   nephiodeployv1alpha1.GroupVersion.Group,
-				Version: nephiodeployv1alpha1.GroupVersion.Version,
-				Kind:    nephiodeployv1alpha1.UPFDeploymentKind,
-			}
-			for _, ref := range dep.Spec.Injectors {
-				/*
-					for _, o := range rl.Items {
-						fn.Logf("configinject resource apiVersion: %s kind: %s\n", o.GetAPIVersion(), o.GetKind())
+			prName := fmt.Sprintf("%s-%s", pr.Spec.RepositoryName, pr.Spec.PackageName)
+
+			latestPR, ok := prmap[prName]
+			if !ok {
+				// we initialize hte prmap independent of the status
+				prmap[prName] = pr
+			} else {
+				// only update if the PR is published
+				if porchv1alpha1.LifecycleIsPublished(pr.Spec.Lifecycle) {
+					if !porchv1alpha1.LifecycleIsPublished(latestPR.Spec.Lifecycle) {
+						// if latest entry in the prmap is not publish but the new pr is published we update the map
+						prmap[prName] = pr
+					} else {
+						// both the latest pr and the new pr are published
+						// update the map with the latest pr
+						// if the revision of the new pr is better than the one of the latest pr in the map
+						if pr.Spec.Revision > latestPR.Spec.Revision {
```

Comment:
"Revision" is a string, and right now anyway it will be "vN" where N is an integer. So, this will result in bogus ordering if there are more than 9 revisions.